pkg/middleware/basic_session.go

```diff
@@ -22,7 +22,19 @@ func NewBasicAuthSessionLoader(validator basic.Validator, sessionGroups []string
 // If no authorization header is found, or the header is invalid, no session
 // will be loaded and the request will be passed to the next handler.
 // If a session was loaded by a previous handler, it will not be replaced.
-func loadBasicAuthSession(validator basic.Validator, sessionGroups []string, next http.Handler) http.Handler {
+func loadBasicAuthSession(validator basic.Validator, sessionGroups []string, preferEmail bool, next http.Handler) http.Handler {
+	// This is a hack to be backwards compatible with the old PreferEmailToUser option.
```

Comment:
Maybe add a `TODO` in the comment so our IDEs can pester us over time so we remember to remove this?